service/src/test/java/bio/terra/pipelines/notifications/NotificationServiceTest.java

```diff
@@ -0,0 +1,289 @@
+import bio.terra.pipelines.common.utils.CommonPipelineRunStatusEnum;
+import bio.terra.pipelines.db.entities.Pipeline;
+import bio.terra.pipelines.db.entities.PipelineRun;
+import bio.terra.pipelines.db.entities.UserQuota;
+import bio.terra.pipelines.db.repositories.PipelineRunsRepository;
+import bio.terra.pipelines.dependencies.rawls.RawlsServiceApiException;
+import bio.terra.pipelines.service.PipelineRunsService;
+import bio.terra.pipelines.service.PipelinesService;
+import bio.terra.pipelines.service.QuotasService;
+import bio.terra.pipelines.testutils.BaseEmbeddedDbTest;
+import bio.terra.pipelines.testutils.TestUtils;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.UUID;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+
+class NotificationServiceTest extends BaseEmbeddedDbTest {
+  @InjectMocks @Autowired NotificationService notificationService;
+  @Autowired PipelineRunsService pipelineRunsService;
+  @Autowired PipelineRunsRepository pipelineRunsRepository;
+  @Autowired PipelinesService pipelinesService;
+  @Autowired QuotasService quotasService;
+  @Autowired NotificationConfiguration notificationConfiguration;
+  @Autowired ObjectMapper objectMapper;
+  @MockBean PubsubService pubsubService;
+  @Mock private FlightContext flightContext;
+
+  UUID testJobId = TestUtils.TEST_NEW_UUID;
+  String testUserId = TestUtils.TEST_USER_ID_1;
+  Integer testQuotaConsumedByJob = 1000;
+  String testUserDescription = TestUtils.TEST_USER_PROVIDED_DESCRIPTION;
+  String testErrorMessage = "test error message";
+
+  @Test
+  void formatDateTime() {
+    Instant instant = Instant.parse("2021-08-25T12:34:56.789Z");
+    String formattedDateTime = notificationService.formatInstantToReadableString(instant);
+    assertEquals("Wed, 25 Aug 2021 12:34:56 GMT", formattedDateTime);
+  }
+
+  @Test
+  void configureAndSendPipelineRunSucceededNotification() throws IOException {
+    Pipeline pipeline = pipelinesService.getPipelineById(1L);
+    PipelineRun writtenPipelineRun =
+        createCompletedPipelineRunInDb(pipeline, CommonPipelineRunStatusEnum.SUCCEEDED);
+
+    // initialize and set user quota
+    UserQuota userQuota =
+        quotasService.getOrCreateQuotaForUserAndPipeline(testUserId, pipeline.getName());
+    UserQuota updatedUserQuota =
+        quotasService.updateQuotaConsumed(userQuota, testQuotaConsumedByJob);
+    int expectedQuotaRemaining = updatedUserQuota.getQuota() - updatedUserQuota.getQuotaConsumed();
+
+    // success is a void method
+    doNothing().when(pubsubService).publishMessage(any(), any(), any());
```

Comment:
can this be made any more specific than `any()`s.  same for the other mocking

Reply:
yes, good call - are you ok leaving the `any()`s when it's mocking throwing an exception?